examples/light-switch-app/silabs/src/BindingHandler.cpp

```diff
@@ -85,6 +86,263 @@ void ProcessOnOffGroupBindingCommand(CommandId commandId, const EmberBindingTabl
     }
 }
 
+void ProcessLevelControlUnicastBindingCommand(BindingCommandData * data, const EmberBindingTableEntry & binding,
+                                              OperationalDeviceProxy * peer_device)
+{
+    auto onSuccess = [](const ConcreteCommandPath & commandPath, const StatusIB & status, const auto & dataResponse) {
+        ChipLogProgress(NotSpecified, "LevelControl command succeeds");
+    };
+
+    auto onFailure = [](CHIP_ERROR error) {
+        ChipLogError(NotSpecified, "LevelControl command failed: %" CHIP_ERROR_FORMAT, error.Format());
+    };
+
+    VerifyOrDie(peer_device != nullptr && peer_device->ConnectionReady());
+
+    switch (data->commandId)
+    {
+    case Clusters::LevelControl::Commands::MoveToLevel::Id:
+    {
+        Clusters::LevelControl::Commands::MoveToLevel::Type moveToLevelCommand;
+        if (auto moveToLevel = std::get_if<BindingCommandData::MoveToLevel>(&data->commandData))
+        {
+            moveToLevelCommand.level           = moveToLevel->level;
+            moveToLevelCommand.transitionTime  = moveToLevel->transitionTime;
+            moveToLevelCommand.optionsMask     = moveToLevel->optionsMask;
+            moveToLevelCommand.optionsOverride = moveToLevel->optionsOverride;
+        }
+        Controller::InvokeCommandRequest(peer_device->GetExchangeManager(), peer_device->GetSecureSession().Value(), binding.remote,
```

Comment:
This should be insinde the if in case the wrong type is passed if i am not mistaken. This applies to all of the cases.